### PR TITLE
[1861] Mobile styles on trainee list

### PR DESCRIPTION
**Pull request by mooktakim**
### Context

https://trello.com/c/tg7twdHw/1861-m-mobile-styles-on-trainee-list-two-breakpoints

### Changes proposed in this pull request

* A bit of organisation of the css classes to allow for 1, 2 and 3 columns on trainees list
* Various breakpoints have been added to match prototype

### Guidance to review

* Go to trainees page, use developer tools to see different responsive sizes.

### CODE REVIEW

**Comment by mooktakim** [comment]
[[M] Mobile styles on trainee list (two breakpoints)](https://trello.com/c/tg7twdHw/1861-m-mobile-styles-on-trainee-list-two-breakpoints)

File: app/webpacker/styles/filter_toggle_button.scss [hunk 1]
```diff
@@ -27,17 +27,26 @@ $shadow-color: #0b0c0c;
...
     display: none;
   }
-}
 
-.filter-toggle-button {
-  float: right;
-  margin-bottom: govuk-spacing(1);
+  .govuk-button {
+    margin: 0;
+  }
```
Comment (edwardhorsford) [inline app/webpacker/styles/filter_toggle_button.scss:39]: I might be cautious about styling on this class, which is an unrelated component.

File: app/webpacker/styles/filter_toggle_button.scss [hunk 2]
```diff
@@ -27,17 +27,26 @@ $shadow-color: #0b0c0c;
...
 
-  @include govuk-media-query($from: desktop) {
-    display: none;
+  @include govuk-media-query($until: 500px) {
+    display: block;
+    float: none;
+    margin-top: 15px;
```
Comment (edwardhorsford) [inline app/webpacker/styles/filter_toggle_button.scss:44]: No magic numbers please.

File: app/webpacker/styles/filter_toggle_button.scss [hunk 3]
```diff
@@ -27,17 +27,26 @@ $shadow-color: #0b0c0c;
...
+  }
+
+  @include govuk-media-query($from: 501px) {
+    margin-top: 20px;
+    margin-bottom: 15px;
```
Comment (edwardhorsford) [inline app/webpacker/styles/filter_toggle_button.scss:49]: No magic numbers please.

File: app/webpacker/styles/filter_toggle_button.scss [hunk 4]
```diff
@@ -27,17 +27,26 @@ $shadow-color: #0b0c0c;
 }
 
 .moj-action-bar__filter {
```
Comment (edwardhorsford) [inline app/webpacker/styles/filter_toggle_button.scss:29]: I know this isn't you, the name of this file does not describe the contents.

File: app/webpacker/styles/filter_toggle_button.scss [hunk 5]
```diff
@@ -27,17 +27,26 @@ $shadow-color: #0b0c0c;
...
+  float: none;
+  margin-top: 30px;
```
Comment (edwardhorsford) [inline app/webpacker/styles/filter_toggle_button.scss:31]: No magic numbers please.

File: app/components/application_record_card/style.scss [hunk 6]
```diff
@@ -90,10 +111,17 @@
...
   span {
-    display: block;
+    @include govuk-media-query($from: 401px) {
+      display: block;
+    }
   }
 }
 
+.app-application-trainee_name {
+  margin-bottom: 5px !important;
```
Comment (edwardhorsford) [inline app/components/application_record_card/style.scss:121]: Does this need `! important`? Also, no magic numbers please.

**Review by edwardhorsford — CHANGES_REQUESTED** [review]
A lot of nit picks to look at.

It would also be good if all classes could follow BEM syntax - at the moment there's a mixed collection of partial BEM and camel-casing.

**Comment by edwardhorsford** [comment]
Comparing to the prototype, I don't think this is working correctly.

In the prototype at below desktop breakpoint (but while filters are visible), the middle column collapses below the first column.

Prototype:
Screenshot 2021-06-18 at 11 46 24


This PR:
Screenshot 2021-06-18 at 11 44 56

File: app/components/application_record_card/style.scss [hunk 7]
```diff
@@ -9,42 +9,61 @@
...
+    // Ensure columns evenly distributed, with no gap at the start or end
+    justify-content: space-between;
+  }
 
-// Trainee column
-.app-application-card_col:nth-last-child(3) {
-  @include govuk-media-query($from: tablet) {
-    width: 40%;
-    padding-right: 15px;
+  .app-application-col-trainee {
```
[the remark this answers is not in the document]
Comment (mooktakim) [reply app/components/application_record_card/style.scss]: It makes it really easy to read the code, adds context to the names.

File: app/components/application_record_card/style.scss [hunk 8]
```diff
@@ -90,10 +126,17 @@
...
   span {
-    display: block;
+    @include govuk-media-query($from: $mobile-small-end) {
+      display: block;
+    }
   }
 }
 
+.app-application-trainee_name {
+  margin-bottom: govuk-spacing(1) !important;
```
Comment (edwardhorsford) [inline app/components/application_record_card/style.scss:136]: We shouldn't need this `! important`. I suspect it's because of the spacing override used on the heading - if we removed that (and wrote styles to achieve a similar thing, it should work.

File: app/components/application_record_card/view.html.erb [hunk 9]
```diff
@@ -1,22 +1,24 @@
...
-  <div class="app-application-card_col">
-    <h<%= heading_level %> class="govuk-heading-m govuk-!-margin-bottom-3">
-      <%= govuk_link_to trainee_name, view_trainee(record) %>
-    </h<%= heading_level %>>
-    <%= trainee_id %>
-    <%= trn %>
-  </div>
+  <div class="app-application-col-trainee-course">
+    <div class="app-application-col-trainee">
+      <h<%= heading_level %> class="govuk-heading-m govuk-!-margin-bottom-3 app-application-trainee_name">
```
Comment (edwardhorsford) [inline app/components/application_record_card/view.html.erb:4]: We can probably remove this margin override which I think your're overriding above anyway.

File: app/components/base.scss [hunk 10]
```diff
@@ -1,4 +1,9 @@
...
 @import "~govuk-frontend/govuk/base";
 
-$moj-images-path: "~@ministryofjustice/frontend/moj/assets/images/";
\ No newline at end of file
+$moj-images-path: "~@ministryofjustice/frontend/moj/assets/images/";
+
+$mobile-small-start: 400px;
+$mobile-small-end: 401px;
+$mobile-big-start: 500px;
+$mobile-big-end: 501px;
```
Comment (edwardhorsford) [inline app/components/base.scss:9]: I might look at how the design system does breakpoints - my hunch is we just need one number per breakpoint, not two.

**Review by edwardhorsford — COMMENTED** [review]
Looking much better. Still a few things I'd tidy up.

Would also be good if the classes followed BEM syntax.

**Comment by edwardhorsford** [comment]
The middle column is slightly more off-centre on the review app than the prototype.

PR
Screenshot 2021-06-24 at 10 19 25

Prototype
Screenshot 2021-06-24 at 10 19 32

Looks like the css is similar so I'm not immediately sure why - but it's 20px to the left. I might guess a different display mode or something since that's the same size as the padding.

File: app/components/application_record_card/style.scss [hunk 11]
```diff
@@ -2,49 +2,85 @@
...
-  @include govuk-media-query($until: tablet) {
+  .app-application-col-status {
     margin-top: govuk-spacing(2);
-    margin-bottom: govuk-spacing(2);
-  }
-  @include govuk-media-query($from: tablet) {
-    width: 25%;
+
+    @include govuk-media-query($from: $mobile-small-end) {
+      width: 40%;
```
Comment (edwardhorsford) [inline app/components/application_record_card/style.scss:69]: I think this needs to be 35% to match the first column.

Comment (mooktakim) [reply app/components/application_record_card/style.scss:69]: This is for size 401px when it's two columns. The first column is 60% and the second is 40%, the same as the prototype.

Comment (edwardhorsford) [reply app/components/application_record_card/style.scss:68]: In the review app the columns are 65% and 40% respectively at desktop breakpoint - this presumably should add to 100%

**Comment by mooktakim** [comment]
[[M] Funding - UI - confirmation page](https://trello.com/c/m9iZr9Jt/1989-m-funding-ui-confirmation-page)

File: app/components/application_record_card/style.scss (hunk 11 above)
Comment (mooktakim) [reply app/components/application_record_card/style.scss:68]: Updated to 35%

**Comment by edwardhorsford** [comment]
I think the main thing remaining on this is consistent use of classes with BEM syntax. The component looks to be called `Application record card`, but then the classes are a mix of `,app-application-card` and `.app-application`

**Review by edwardhorsford — APPROVED** [review]
Looking great! 🎉

File: yarn.lock [hunk 12]
```diff
@@ -2582,9 +2582,9 @@ caniuse-api@^3.0.0:
     lodash.uniq "^4.5.0"
 
 caniuse-lite@^1.0.0, caniuse-lite@^1.0.30000981, caniuse-lite@^1.0.30001109, caniuse-lite@^1.0.30001219:
-  version "1.0.30001236"
-  resolved "https://registry.yarnpkg.com/caniuse-lite/-/caniuse-lite-1.0.30001236.tgz#0a80de4cdf62e1770bb46a30d884fc8d633e3958"
-  integrity sha512-o0PRQSrSCGJKCPZcgMzl5fUaj5xHe8qA2m4QRvnyY4e1lITqoNkr7q/Oh1NcpGSy0Th97UZ35yoKcINPoq7YOQ==
+  version "1.0.30001239"
```
Comment (AbigailMcP) [inline yarn.lock:2585]: Can we revert this change so that it's `yarnpkg` like the rest of them?

**Review by AbigailMcP — APPROVED** [review]
Just one minor thing then lgtm! 🚢

File: yarn.lock (hunk 12 above)
Comment (mooktakim) [reply yarn.lock]: How did that get in there, reverted
